slashing-protection/src/test/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidatorTest.java

```diff
@@ -64,6 +68,36 @@ void verifyCachedGVRIsUsedForNewGVR(final Jdbi jdbi) {
     verify(metadataDao, times(1)).insertGenesisValidatorsRoot(any(), eq(gvr));
   }
 
+  @Test
+  void verifyCachedGVRReturnsTrueFromMultipleThreads(final Jdbi jdbi) throws InterruptedException {
+    var gvrValidator = new GenesisValidatorRootValidator(jdbi, metadataDao);
+    var gvr = Bytes32.leftPad(Bytes.of(3));
+
+    var numberOfThreads = 10;
+    var executorService = Executors.newFixedThreadPool(numberOfThreads);
+    var allCachedGVRMatches = new AtomicBoolean(true);
+    for (int i = 0; i < numberOfThreads; i++) {
+      executorService.submit(
+          () -> {
+            boolean gvrMatches = gvrValidator.checkGenesisValidatorsRootAndInsertIfEmpty(gvr);
+            if (!gvrMatches) {
+              allCachedGVRMatches.set(false);
+            }
+          });
+    }
+
+    // Shutdown the executor service, no new tasks will be accepted
+    executorService.shutdown();
+
+    // wait for all threads to finish
+    var successfulTerminate = executorService.awaitTermination(5, MINUTES);
```

Comment:
Could probably even make this SECONDS so we don't slow down the tests too much if this were to happen

Reply:
Reduced it to 1 minute.